bundles/org.openhab.binding.warmup/src/main/java/org/openhab/binding/warmup/internal/handler/MyWarmupAccountHandler.java

```diff
@@ -95,15 +95,12 @@ public void cancelRefresh() {
     public synchronized void refreshFromServer() {
         try {
             queryResponse = api.getStatus();
+            updateStatus(ThingStatus.ONLINE);
         } catch (MyWarmupApiException e) {
             queryResponse = null;
-            logger.debug("{}", e.getMessage(), e);
+            logger.debug("{}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.